src/cargo/ops/cargo_compile/unit_generator.rs

```diff
@@ -267,7 +268,21 @@ impl<'a> UnitGenerator<'a, '_> {
                 })
                 .collect::<Vec<_>>();
             let suggestion = closest_msg(target_name, targets.iter(), |t| t.name(), "target");
-            if !suggestion.is_empty() {
+            let target_elsewhere = if is_glob {
+                let pattern = build_glob(target_name)?;
+                let filter = |t: &Target| is_expected_kind(t) && pattern.matches(t.name());
+                self.get_targets_from_other_packages(filter)?
+            } else {
+                let filter = |t: &Target| t.name() == target_name && is_expected_kind(t);
+                self.get_targets_from_other_packages(filter)?
+            };
```

Comment:
Would it be worth extracting the general pattern of filter creation rather than duplicate it?